compiler/semmagic.nim

```diff
@@ -378,8 +378,11 @@ proc magicsAfterOverloadResolution(c: PContext, n: PNode,
   of mArrPut:
     result = semArrPut(c, n, flags)
   of mAsgn:
-    if n[0].sym.name.s == "=":
+    case n[0].sym.name.s.normalize
```

Comment:
``=destroy`` is not mapped to ``mAsgn`` anymore, so you can leave the condition at ``if n[0].sym.name.s == "="`` which is faster.

Reply:
```diff
diff --git a/compiler/semmagic.nim b/compiler/semmagic.nim
index 54f67ae99..df2c084a1 100644
--- a/compiler/semmagic.nim
+++ b/compiler/semmagic.nim
@@ -378,13 +378,10 @@ proc magicsAfterOverloadResolution(c: PContext, n: PNode,
   of mArrPut:
     result = semArrPut(c, n, flags)
   of mAsgn:
-    case n[0].sym.name.s.normalize
-    of "=":
+    if n[0].sym.name.s == "=":
       result = semAsgnOpr(c, n)
-    of "=sink":
-      result = semShallowCopy(c, n, flags)
     else:
-      result = n
+      result = semShallowCopy(c, n, flags)
   of mIsPartOf: result = semIsPartOf(c, n, flags)
   of mTypeTrait: result = semTypeTraits(c, n)
   of mAstToStr:
```

I believe this is what you meant?

Reply:
Exactly.